src/coreclr/jit/valuenum.cpp

```diff
@@ -18,17 +18,6 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 #include "valuenum.h"
```

Comment:
`valuenum.cpp` is now calling a lot of C runtime functions directly and that means that we lose out on any handling that the PAL wrappers were covering.

We should likely explicitly validate that scenarios which had special PAL handling are still correctly handled here via constant folding.

This likely, at a minimum, should entail us running https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime/tests/System.Runtime.Extensions.Tests/System/MathF.cs and https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime/tests/System.Runtime.Extensions.Tests/System/Math.cs, in both regular and `TC=0` modes to ensure that it produces the correct results.

It may have changed since, but when I added all the validation/coverage to ensure we were IEEE 754 compliant; there were various scenarios where some platforms did the wrong thing, so the concern is around such errors slipping back in.

Reply:
Given that we've been using the CRT functionality directly with Mono for quite a while, I'm not too concerned, at least for platforms that we regularly run the libraries tests on Mono for (such as MacOS/iOS/tvOS/Android). For other platforms, I can understand the desire for extra validation.

For many of the tests

Reply:
👍. Just mostly concerned about us regressing this scenario where a lot of effort has been put into moving us towards IEEE 754 spec correctness.